yt_dlp/extractor/n1.py

```diff
@@ -0,0 +1,88 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .youtube import YoutubeIE
+from .common import InfoExtractor
+from ..utils import (
+    unified_timestamp,
+    extract_attributes,
+)
+
+
+class N1InfoAssetIE(InfoExtractor):
+    _VALID_URL = r'https?://best-vod\.umn\.cdn\.united\.cloud/stream\?asset=(?P<id>[^&]+)'
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        formats = self._extract_m3u8_formats(
+            url, video_id, 'mp4', entry_protocol='m3u8_native', m3u8_id='hls', fatal=False)
+
+        self._sort_formats(formats)
+
+        return {
+            'id': video_id,
+            'title': video_id,
+            'formats': formats
+        }
+
+
+class N1InfoIIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:(?:ba|rs|hr)\.)?n1info\.(?:com|si)/(?:[^/]+/){1,2}(?P<id>[^/]+)'
+    _TESTS = [{
+        'url': 'https://rs.n1info.com/sport-klub/tenis/kako-je-djokovic-propustio-istorijsku-priliku-video/',
+        'md5': '01ddb6646d0fd9c4c7d990aa77fe1c5a',
+        'info_dict': {
+            'id': 'L5Hd4hQVUpk',
+            'ext': 'mp4',
+            'upload_date': '20210913',
+            'title': 'Ozmo i USO21, ep. 13: Novak Đoković – Danil Medvedev | Ključevi Poraza, Budućnost | SPORT KLUB TENIS',
+            'description': 'md5:467f330af1effedd2e290f10dc31bb8e',
+            'uploader': 'Sport Klub',
+            'uploader_id': 'sportklub',
+        }
+    }, {'url': 'https://rs.n1info.com/vesti/djilas-los-plan-za-metro-nece-resiti-nijedan-saobracajni-problem/',
+        'info_dict': {
+            'id': 'bgmetrosot2409zta20210924174316682-n1info-rs-worldwide',
+            'ext': 'mp4',
+            'title': 'bgmetrosot2409zta20210924174316682-n1info-rs-worldwide'
+        }},
+        {'url': 'https://n1info.si/novice/slovenija/zadnji-dnevi-na-kopaliscu-ilirija-ilirija-ni-umrla-ubili-so-jo/',
+         'info_dict': {
+             'id': 'ljsottomazilirija3060921-n1info-si-worldwide',
+             'ext': 'mp4',
+             'title': 'ljsottomazilirija3060921-n1info-si-worldwide'
+         }
+         }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(r'<h1[^>]+>(.+?)</h1>', webpage, 'title')
+        timestamp = unified_timestamp(self._html_search_meta('article:published_time', webpage))
+
+        videos = re.findall(r'(?m)(<video[^>]+>)', webpage)
+        entries = []
+        for video in videos:
+            video_data = extract_attributes(video)
+            entries.append(self.url_result(
+                video_data.get('data-url'), ie=N1InfoAssetIE.ie_key(),
+                video_id=video_data.get('id'), video_title=title))
```

Comment:
If I already specified the video title, why does the extractor have to return it?

Reply:
`url_result` creates `_type=url` entry. The data extracted here discarded once the video extractor runs. If you want to carry that data, you must return `{'_type': 'url_transparent', 'id': id, ...}` instead

Reply:
I still get errors
```
$ python -m yt_dlp -v https://n1info.si/novice/slovenija/zadnji-dnevi-na-kopaliscu-ilirija-ilirija-ni-umrla-ubili-so-jo/
[debug] User config file: /home/pogaca/.config/yt-dlp/config
[debug] User config: ['-f', 'mkv', '-f', 'bestvideo[height<=1080]+bestaudio/best/standard', '-o', '~/Downloads/%(title)s - %(id)s.%(ext)s']
[debug] Command-line config: ['-v', 'https://n1info.si/novice/slovenija/zadnji-dnevi-na-kopaliscu-ilirija-ilirija-ni-umrla-ubili-so-jo/']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] yt-dlp version 2021.09.25 (source)
[debug] Plugin Extractors: ['SamplePlugin']
[debug] Git HEAD: 2644e2123
[debug] Python version 3.9.7 (CPython 64bit) - Linux-5.14.7-arch1-1-x86_64-with-glibc2.33
[debug] exe versions: ffmpeg 4.4, ffprobe 4.4, rtmpdump 2.4
[debug] Optional libraries: Crypto, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [N1InfoI] Extracting URL: https://n1info.si/novice/slovenija/zadnji-dnevi-na-kopaliscu-ilirija-ilirija-ni-umrla-ubili-so-jo/
[N1InfoI] zadnji-dnevi-na-kopaliscu-ilirija-ilirija-ni-umrla-ubili-so-jo: Downloading webpage
[download] Downloading playlist: Zadnji dnevi na kopališču Ilirija: “Ilirija ni umrla, ubili so jo”
[N1InfoI] playlist Zadnji dnevi na kopališču Ilirija: “Ilirija ni umrla, ubili so jo”: Collected 2 videos; downloading 2 of them
[download] Downloading video 1 of 2
[debug] [N1InfoAsset] Extracting URL: https://best-vod.umn.cdn.united.cloud/stream?asset=ljsottomazilirija3060921-n1info-si-worldwide&stream=hp1400&t=0&player=m3u8v&sp=n1info&u=n1info&p=n1Sh4redSecre7iNf0
[N1InfoAsset] ljsottomazilirija3060921-n1info-si-worldwide: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
ERROR: Missing "id" field in extractor result; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1227, in wrapper
    return func(self, *args, **kwargs)
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1599, in __process_iterable_entry
    return self.process_ie_result(
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1367, in process_ie_result
    return self.process_ie_result(
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1315, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 2092, in process_video_result
    raise ExtractorError('Missing "id" field in extractor result')
yt_dlp.utils.ExtractorError: Missing "id" field in extractor result; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.

[download] Downloading video 2 of 2
[debug] [N1InfoAsset] Extracting URL: https://best-vod.umn.cdn.united.cloud/stream?asset=ljsottomailirija2060921-n1info-si-worldwide&stream=hp1400&t=0&player=m3u8v&sp=n1info&u=n1info&p=n1Sh4redSecre7iNf0
[N1InfoAsset] ljsottomailirija2060921-n1info-si-worldwide: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
ERROR: Missing "id" field in extractor result; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1227, in wrapper
    return func(self, *args, **kwargs)
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1599, in __process_iterable_entry
    return self.process_ie_result(
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1367, in process_ie_result
    return self.process_ie_result(
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1315, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 2092, in process_video_result
    raise ExtractorError('Missing "id" field in extractor result')
yt_dlp.utils.ExtractorError: Missing "id" field in extractor result; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.

[download] Finished downloading playlist: Zadnji dnevi na kopališču Ilirija: “Ilirija ni umrla, ubili so jo”
```
Changes I have made
```
diff --git a/yt_dlp/extractor/n1.py b/yt_dlp/extractor/n1.py
index 36d1b5096..4425288a8 100644
--- a/yt_dlp/extractor/n1.py
+++ b/yt_dlp/extractor/n1.py
@@ -30,11 +30,7 @@ class N1InfoAssetIE(InfoExtractor):
 
         self._sort_formats(formats)
 
-        return {
-            'id': video_id,
-            'title': video_id,
-            'formats': formats
-        }
+        return {'formats': formats}
 
 
 class N1InfoIIE(InfoExtractor):
@@ -81,9 +77,13 @@ class N1InfoIIE(InfoExtractor):
         entries = []
         for video in videos:
             video_data = extract_attributes(video)
-            entries.append(self.url_result(
-                video_data.get('data-url'), ie=N1InfoAssetIE.ie_key(),
-                video_id=video_data.get('id'), video_title=title))
+            entries.append({
+                '_type': 'url_transparent',
+                'id': video_data.get('id'),
+                'url': video_data.get('data-url'),
+                'title': title,
+                'timestamp': timestamp,
+                'ie_key': N1InfoAssetIE.ie_key()})
 
         youtube_videos = re.findall(r'(<iframe[^>]+>)', webpage)
         for youtube_video in youtube_videos:

```

Reply:
I still get errors 
Am I doing something wrong?
```
$ python -m yt_dlp -v https://n1info.si/novice/slovenija/zadnji-dnevi-na-kopaliscu-ilirija-ilirija-ni-umrla-ubili-so-jo/
[debug] User config file: /home/pogaca/.config/yt-dlp/config
[debug] User config: ['-f', 'mkv', '-f', 'bestvideo[height<=1080]+bestaudio/best/standard', '-o', '~/Downloads/%(title)s - %(id)s.%(ext)s']
[debug] Command-line config: ['-v', 'https://n1info.si/novice/slovenija/zadnji-dnevi-na-kopaliscu-ilirija-ilirija-ni-umrla-ubili-so-jo/']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] yt-dlp version 2021.09.25 (source)
[debug] Plugin Extractors: ['SamplePlugin']
[debug] Git HEAD: 2644e2123
[debug] Python version 3.9.7 (CPython 64bit) - Linux-5.14.7-arch1-1-x86_64-with-glibc2.33
[debug] exe versions: ffmpeg 4.4, ffprobe 4.4, rtmpdump 2.4
[debug] Optional libraries: Crypto, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [N1InfoI] Extracting URL: https://n1info.si/novice/slovenija/zadnji-dnevi-na-kopaliscu-ilirija-ilirija-ni-umrla-ubili-so-jo/
[N1InfoI] zadnji-dnevi-na-kopaliscu-ilirija-ilirija-ni-umrla-ubili-so-jo: Downloading webpage
[download] Downloading playlist: Zadnji dnevi na kopališču Ilirija: “Ilirija ni umrla, ubili so jo”
[N1InfoI] playlist Zadnji dnevi na kopališču Ilirija: “Ilirija ni umrla, ubili so jo”: Collected 2 videos; downloading 2 of them
[download] Downloading video 1 of 2
[debug] [N1InfoAsset] Extracting URL: https://best-vod.umn.cdn.united.cloud/stream?asset=ljsottomazilirija3060921-n1info-si-worldwide&stream=hp1400&t=0&player=m3u8v&sp=n1info&u=n1info&p=n1Sh4redSecre7iNf0
[N1InfoAsset] ljsottomazilirija3060921-n1info-si-worldwide: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
ERROR: Missing "id" field in extractor result; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1227, in wrapper
    return func(self, *args, **kwargs)
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1599, in __process_iterable_entry
    return self.process_ie_result(
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1367, in process_ie_result
    return self.process_ie_result(
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1315, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 2092, in process_video_result
    raise ExtractorError('Missing "id" field in extractor result')
yt_dlp.utils.ExtractorError: Missing "id" field in extractor result; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.

[download] Downloading video 2 of 2
[debug] [N1InfoAsset] Extracting URL: https://best-vod.umn.cdn.united.cloud/stream?asset=ljsottomailirija2060921-n1info-si-worldwide&stream=hp1400&t=0&player=m3u8v&sp=n1info&u=n1info&p=n1Sh4redSecre7iNf0
[N1InfoAsset] ljsottomailirija2060921-n1info-si-worldwide: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
ERROR: Missing "id" field in extractor result; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1227, in wrapper
    return func(self, *args, **kwargs)
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1599, in __process_iterable_entry
    return self.process_ie_result(
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1367, in process_ie_result
    return self.process_ie_result(
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 1315, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/home/pogaca/Documents/yt-dlp-n1/yt_dlp/YoutubeDL.py", line 2092, in process_video_result
    raise ExtractorError('Missing "id" field in extractor result')
yt_dlp.utils.ExtractorError: Missing "id" field in extractor result; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
```

Code:
```

diff --git a/yt_dlp/extractor/n1.py b/yt_dlp/extractor/n1.py
index 36d1b5096..e55d79bb2 100644
--- a/yt_dlp/extractor/n1.py
+++ b/yt_dlp/extractor/n1.py
@@ -30,11 +30,7 @@ class N1InfoAssetIE(InfoExtractor):
 
         self._sort_formats(formats)
 
-        return {
-            'id': video_id,
-            'title': video_id,
-            'formats': formats
-        }
+        return {'formats': formats}
 
 
 class N1InfoIIE(InfoExtractor):
@@ -81,9 +77,13 @@ class N1InfoIIE(InfoExtractor):
         entries = []
         for video in videos:
             video_data = extract_attributes(video)
-            entries.append(self.url_result(
-                video_data.get('data-url'), ie=N1InfoAssetIE.ie_key(),
-                video_id=video_data.get('id'), video_title=title))
+            entries.append({
+                '_type': 'url_transparent',
+                'url': video_data.get('data-url'),
+                'id': video_data.get('id'),
+                'title': title,
+                'timestamp': timestamp,
+                'ie_key': N1InfoAssetIE.ie_key()})
 
         youtube_videos = re.findall(r'(<iframe[^>]+>)', webpage)
         for youtube_video in youtube_videos:
```

Reply:
u still need to return the id in the assetIE

Reply:
actually, return a title also in assetIE, just in case someone uses the asset url directly